core/src/main/java/brooklyn/catalog/internal/BasicBrooklynCatalog.java

```diff
@@ -750,7 +780,37 @@ private String setFromItemIfUnset(String oldValue, Map<?,?> item, String fieldAt
         return oldValue;
     }
 
+    private Collection<CatalogItemDtoAbstract<?, ?>> scanAnnotationsFromLocal(ManagementContext mgmt) {
+        CatalogDto dto = CatalogDto.newNamedInstance("Local Scanned Catalog", "All annotated Brooklyn entities detected in the classpath", "scanning-local-classpath");
+        return scanAnnotationsInternal(mgmt, new CatalogDo(dto));
+    }
     
+    private Collection<CatalogItemDtoAbstract<?, ?>> scanAnnotationsFromBundles(ManagementContext mgmt, Collection<CatalogBundle> libraries) {
+        String[] urls = null;
+        CatalogDto dto = CatalogDto.newNamedInstance("Bundles Scanned Catalog", "All annotated Brooklyn entities detected in the classpath", "scanning-bundles-classpath-"+libraries.hashCode());
+        urls = new String[libraries.size()];
+        int i=0;
+        for (CatalogBundle b: libraries)
+            urls[i++] = b.getUrl();
+            
+        CatalogDo subCatalog = new CatalogDo(dto);
+        subCatalog.addToClasspath(urls);
```

Comment:
I see that below you are handling the `null` case, but then if all libraries are name-value pairs, the code path will go in the "scanning catalog root classpath" case. Better move the null check here.